openhtf/util/data.py

```diff
@@ -158,7 +158,7 @@ def convert_to_base_types(obj, ignore_keys=tuple(), tuple_type=tuple):
     return float(obj)
 
   # Convert all other types to strings.
-  return str(obj)
+  return unicode(obj)
```

Comment:
Do we need to convert to unicode? There are cases where the data is actually bytes (serial numbers, measurement values, etc) or doing this potentially expensive conversion is unwanted. Can we split this into a 3-way switch? If it's a str, return it, if it's unicode, return that, and then actually return unicode(obj)?

Reply:
I don't think I understand the code well enough right now to answer that, but what you say makes sense.

Reply:
Did you mean to change this later then?

Reply:
If I get the time, yes.

Reply:
It turns out unicode strings are already skipped because they're listed in `PASSTHROUGH_TYPES`. I've rolled this part back.